### ISSUE TITLE
[SC3003] `$'..'` is POSIX.1-2024 compliant

### ISSUE THREAD

**Issue by Neved4** [koalaman/shellcheck#3098]
#### For bugs
- Rule Id: `SC3003`:
- My shellcheck version: `online`
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
export var=$'\033[0m'
```

#### Here's what shellcheck currently says:

```console
export VAR=$'\e[1;32m'
           ^-- [SC3003](https://www.shellcheck.net/wiki/SC3003) (warning): In POSIX sh, $'..' is undefined.
```

#### Here's what I wanted or expected to see:

Believe this could be a false positive. [POSIX.1-2024](https://pubs.opengroup.org/onlinepubs/9799919799/) seems to have added this, under [§2.2.4 Dollar-Single-Quotes](https://pubs.opengroup.org/onlinepubs/9799919799/).

It can also be found in Austin Group's Bug Tracker: [0000249: Add standard support for $'...' in shell - Austin Group Defect Tracker](https://austingroupbugs.net/view.php?id=249).

Ideally, the current warning should disappear for POSIX shells. Its functionality could be preserved in an optional flag to work with legacy shells.

**Comment by krushia** [comment]
See also #1678

**Comment by Neved4** [comment]
If it's a dupe feel free to close either one in favor of the other, much love ✌🏻
